models/User.js

```diff
@@ -105,16 +105,17 @@ UserSchema.statics.authenticate = function(username, password) {
 			if(!user) {
 				reject({message: 'Username does not exist'});
 			}
-
-			// Match password
-			bcrypt.compare(password, user.password, (err, isMatch) => {
-				if (isMatch && !err) {
-					resolve(user);
-				}
-				else {
-					reject({message: 'Password is incorrect'});
-				}
-			});
+			else {
```

Comment:
Fine, but was ok as it was - this `reject` is like a return statement in a function, anything below it won't be reached! (If this was broken login wouldn't have been working, like, ever :P)

Reply:
Reject permanently sets the resolution state, but it doesn't return the function (see https://stackoverflow.com/questions/32536049/do-i-need-to-return-after-early-resolve-reject). Login still works fine because the resolution state is already set and will not be changed by the following resolves or rejects. It doesn't have any real effect, but does prevent some code from running that doesn't need to run.

Reply:
Mm cool, my mistake - happy to have it be more explicit / save a fraction of a second or two :+1: